test/integration/ctl/ctl_create_cr_test.go

```diff
@@ -288,31 +422,31 @@ func TestCtlCreateCR(t *testing.T) {
 				t.Errorf("CR created in unexpected Namespace, expected: %s, actual: %s", test.expNamespace, gotCr.Namespace)
 			}
 
-			// Check the file where the private key is stored
-			keyData, err := ioutil.ReadFile(test.expKeyFilename)
-			if err != nil {
-				t.Errorf("error when reading file storing private key: %v", err)
-			}
-			_, err = pki.DecodePrivateKeyBytes(keyData)
-			if err != nil {
-				t.Errorf("invalid private key: %v", err)
-			}
-
 			// If applicable, check the file where the certificate is stored
-			if test.fetchCert {
+			// If the expected error message is the one below, we skip checking
+			// because no certificate will have been written to file
+			if test.fetchCert && test.expErrMsg != "error when waiting for CertificateRequest to be signed: timed out waiting for the condition" {
 				certData, err := ioutil.ReadFile(test.expCertFilename)
 				if err != nil {
 					t.Errorf("error when reading file storing private key: %v", err)
 				}
 
 				if !bytes.Equal(test.expCertFileContent, certData) {
-					t.Errorf("certificate written to file is wrong, expected: %v,\nactual: %v", test.expCertFileContent, certData)
+					t.Errorf("certificate written to file is wrong, expected: %s,\nactual: %s", test.expCertFileContent, certData)
 				}
 			}
 		})
 	}
 }
 
+func getTestDataDir(t *testing.T) string {
+	testWorkingDirectory, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return testWorkingDirectory + "/testdata/"
```

Comment:
we have `path.Join()` in Go which does this for you with working around weird edge causes (cough windows)